packages/snap/src/utils/types.ts

```diff
@@ -1,17 +1,32 @@
 import * as t from 'io-ts'
 
-export const SimpleTransaction = t.intersection([
-    t.type({
-        to: t.string,
-        value: t.string
-    }),
-    t.partial({
-        feeCurrency: t.union([t.string, t.undefined])
-    })
+export const TransactionConfig = t.partial({
+    from: t.union([t.string, t.number]),
+    to: t.string,
+    value: t.union([t.number, t.string, t.bigint]),
+    gas: t.union([t.string, t.number]),
+    gasPrice: t.union([t.number, t.string, t.bigint]),
+    data: t.string,
+    nonce: t.number,
+    chainId: t.string,
+    common: t.string,
+    chain: t.string,
+    hardfork: t.string
```

Comment:
when submitting a transaction ive never seen hardfork as a parameter

Reply:
Hi!, yes i agree most of these are not needed, but i copied them over here as best as i can from here 
https://github.com/web3/web3.js/blob/1.x/packages/web3-core/types/index.d.ts#L127

Reply:
ok.  I see these are eventually passed into `wallet.sendTransaction()` from celo-ethers-wrapper which  defines a `Celotransaction`  that extends ethers' `TransactinoRequest` https://github.com/ethers-io/ethers.js/blob/5210b68a7837654c6b84207a45e1e573d9472d1a/src.ts/providers/provider.ts#L124C18-L124C36

seems this would be abetter thing to model the io-ts type after. as it is the thing doing the actual signing and will need certain params.

Reply:
Thank you so much!
Just merged Alecs PR into this branch, would you mind looking one more time?